lib/srv/db/access_test.go

```diff
@@ -814,6 +815,13 @@ func TestAccessMongoDB(t *testing.T) {
 						// Create user/role with the requested permissions.
 						testCtx.createUserAndRole(ctx, t, test.user, test.role, test.allowDbUsers, test.allowDbNames)
 
+						// Mongo driver establish servers connection to the upstream mongo server
+						// To decrease the deal during dialing Mongo TestServer by Engine wait for
+						// PrecomputeKeys.
+						require.Eventually(t, func() bool {
+							return native.PrecomputeKeysCount() > 3
+						}, time.Minute, time.Second, "failed to wait for pre-computed keys ")
```

Comment:
I worry this will continue to be flaky, good chance some other test immediately takes those keys. In fact, just within `TestAccessMongoDB`, we're going to spin up 48 parallel tests, all will wait for just 3 precomputed keys, and then all will continue and start using up those keys.

Is there some other way to use a static list of precomputed keys, or otherwise increase the delay that will be tolerated?